control-plane-operator/controllers/awsprivatelink/awsprivatelink_controller.go

```diff
@@ -286,6 +287,11 @@ func (r *AWSEndpointServiceReconciler) Reconcile(ctx context.Context, req ctrl.R
 	}
 	hcp := &hcpList.Items[0]
 
+	if isPaused, duration := supportutil.IsReconciliationPaused(log, hcp.Spec.PausedUntil); isPaused {
+		log.Info("Reconciliation paused", "pausedUntil", *hcp.Spec.PausedUntil)
```

Comment:
pausing here is not consistent with what we do in HostedCluster and NodePool where we keep reconciling status.

Reply:
I've tested it in local and the behaviour it's the expected one. For the testcase it's basically: 
- Deploy a private cluster
- Put it in pause once deployed
- Delete the AWSEndPointService and the Service from the HCP namespace
- And wait for a reconciliation, the result it's that they should not be recreated
- Unpause it and wait for recreation.

If I put this "pause" in a more advanced place, it will try to CreateOrUpdate the resources we don't want to be recreated when it's paused.

Reply:
>If I put this "pause" in a more advanced place, it will try to CreateOrUpdate the resources we don't want to be recreated when it's paused.

@jparrill Pause behaviour must be consistent across resources unless there's a reason to not.
It doesn't need to go in a "more advanced" place, if achieving that consistency might require refactor this controller code to split status and spec reconciliation in the same we do in the HC/NodePool we need to capture that in a follow up ticket.
So either we need a TODO in this PR with jira ticket reference here to be consistent in a follow up or a comment in this PR explaining the limitations that would prevent us from ever be consistent.